commcare/src/main/java/org/motechproject/commcare/service/CommcareSchemaService.java

```diff
@@ -43,4 +43,18 @@
      * @return the list of matching applications
      */
     List<CommcareApplicationJson> retrieveApplications(String configName);
+    
+    /**
+     * Retrieves case types with application name
+     * @return Map of case types. Key represents a case type, while a value is a
+     *         application name the case type belongs to.
+     */
+    Map<String, String> getCaseTypesToApplicationName(String configName);
```

Comment:
Instead of creating some weird maps, can we just create a separate model class, that will keep the case and the application name? It will be much cleaner and easier to understand. Moreover, can you rename the methods to getCaseTypesWithApplicationName?